router/user.go

```diff
@@ -36,17 +43,26 @@ func (rt *Router) handleLoginSave(w http.ResponseWriter, r *http.Request) {
 			"One or more fields was not submitted. Please try again.")
 	}
 
-	_, err := rt.UserRepo.GetByCredentials(username, password)
+	u, err := rt.UserRepo.GetByCredentials(username, password)
 	if err != nil {
 		displayErrorPage(w, r, http.StatusNotFound,
 			"We found no user with the provided credentials in the database. Please check your username and password, and try again.")
 	}
 
+	session, err := rt.Store.Get(r, "session")
+	// fmt.Print("Session: %v\n", session)
+	session.Values["userID"] = u.ID
+	session.Save(r, w)
+
 	// TODO: Set session before redirecting and get user's actual active budget
 	http.Redirect(w, r, "/dashboard?id=1", http.StatusSeeOther)
 }
 
 func (rt *Router) handleRegister(w http.ResponseWriter, r *http.Request) {
+	if rt.userIsLoggedIn(w, r) {
```

Comment:
This and the one below can be removed; a logged in user would get redirected earlier, and duplicate entries are denied anyway due to uniqueness constraints.